lib/postqueue/queue/enqueue.rb

```diff
@@ -0,0 +1,26 @@
+module Postqueue
+  class Queue
+    def enqueue(op:, entity_id:, channel: nil)
+      check_channel_support!(channel)
+
+      # enqueue items. If the operation is idempotent, we can ignore any operation
+      # that is already queued, since in that case the new operation's effect will
+      # only replicate what the already queued operation is doing, without any
+      # discernible effect.
+      enqueued_item_count = item_class.enqueue op: op, entity_id: entity_id, channel: channel,
+                                               ignore_if_exists: Postqueue.callback(op: op)&.idempotent?
+      return 0 if enqueued_item_count == 0
```

Comment:
This looks weird, since idempotent? returns a bool but we should have a int instead